src/app/parcels/parcelsTable/ParcelsTable.tsx

```diff
@@ -0,0 +1,383 @@
+import { DbParcelRow } from "@/databaseUtils";
+import { DateRangeState } from "@/components/DateInputs/DateRangeInputs";
+import {
+    defaultNumberOfParcelsPerPage,
+    numberOfParcelsPerPageOptions,
+    parcelIdParam,
+} from "@/app/parcels/parcelsTable/constants";
+import {
+    defaultShownHeaders,
+    parcelTableHeaderKeysAndLabels,
+    toggleableHeaders,
+} from "@/app/parcels/parcelsTable/headers";
+import {
+    getClientIdAndIsActiveErrorMessage,
+    getParcelDataErrorMessage,
+    parcelTableColumnDisplayFunctions,
+} from "@/app/parcels/parcelsTable/format";
+import { parcelTableColumnStyleOptions } from "@/app/parcels/parcelsTable/styles";
+import parcelsSortableColumns, {
+    defaultParcelsSortConfig,
+} from "@/app/parcels/parcelsTable/sortableColumns";
+import { useCallback, useEffect, useMemo, useRef, useState } from "react";
+import dayjs from "dayjs";
+import {
+    shouldFilterBeDisabled,
+    shouldParcelBePacked,
+} from "@/app/parcels/parcelsTable/packingManagerHelpers";
+import { AppRouterInstance } from "next/dist/shared/lib/app-router-context.shared-runtime";
+import { useSearchParams } from "next/navigation";
+import {
+    getClientIdAndIsActive,
+    getParcelIds,
+    getParcelsDataAndCount,
+} from "@/app/parcels/parcelsTable/fetchParcelTableData";
+import supabase from "@/supabaseClient";
+import { searchForBreakPoints } from "@/app/parcels/parcelsTable/conditionalStyling";
+import { subscriptionStatusRequiresErrorMessage } from "@/common/subscriptionStatusRequiresErrorMessage";
+
+interface ParcelsTableProps {
+    setSelectedParcelId: (parcelId: string | null) => void;
+    setSelectedClientDetails: (clientDetails: SelectedClientDetails | null) => void;
+    checkedParcelIds: string[];
+    setCheckedParcelIds: (ids: string[]) => void;
+    setModalIsOpen: (isOpen: boolean) => void;
+    router: AppRouterInstance;
+    sortState: ParcelsSortState;
+    setSortState: (sortState: ParcelsSortState) => void;
+    primaryFilters: (
+        | ParcelsFilter<string>
+        | ParcelsFilter<DateRangeState>
+        | ParcelsFilter<string[]>
+    )[];
+    setPrimaryFilters: (
+        filters: (ParcelsFilter<string> | ParcelsFilter<DateRangeState> | ParcelsFilter<string[]>)[]
+    ) => void;
+    additionalFilters: (
+        | ParcelsFilter<string>
+        | ParcelsFilter<DateRangeState>
+        | ParcelsFilter<string[]>
+    )[];
+    setAdditionalFilters: (
+        filters: (ParcelsFilter<string> | ParcelsFilter<DateRangeState> | ParcelsFilter<string[]>)[]
+    ) => void;
+    areFiltersLoadingForFirstTime: boolean;
+    setErrorMessage: (errorMessage: string | null) => void;
+    setModalErrorMessage: (errorMessage: string | null) => void;
+    isPackingManagerView: boolean;
+}
+
+const ParcelsTable: React.FC<ParcelsTableProps> = ({
+    setSelectedParcelId,
+    setSelectedClientDetails,
+    checkedParcelIds,
+    setCheckedParcelIds,
+    setModalIsOpen,
+    router,
```

Comment:
We can get the router within the component with `useRouter` rather than passing it through as a prop. I know that means we'll need `useRouter` in a couple of places, but I think that's cleaner than passing it as props.